CriticalMass/MapViewController.swift

```diff
@@ -186,15 +229,42 @@ extension MapViewController: MKMapViewDelegate {
     // MARK: MKMapViewDelegate
 
     func mapView(_ mapView: MKMapView, viewFor annotation: MKAnnotation) -> MKAnnotationView? {
-        guard annotation is MKUserLocation == false else {
+        guard let userAnnotation = annotation as? IdentifiableAnnnotation else {
+            debugPrint("⚠️ Not a UserAnnotation.")
             return nil
         }
+        var view: MKAnnotationView
+        switch userAnnotation.type {
+        case .friend:
+            if #available(iOS 11.0, *) {
+                if let dequedView = mapView.dequeueReusableAnnotationView(withIdentifier: FriendAnnotationView.reuseIdentifier, for: userAnnotation) as? FriendAnnotationView {
+                    dequedView.annotation = userAnnotation
+                    view = dequedView
+                } else {
+                    view = FriendAnnotationView(annotation: userAnnotation,
+                                                reuseIdentifier: FriendAnnotationView.reuseIdentifier)
+                }
+            } else {
+                view = FriendAnnotationView(annotation: userAnnotation,
+                                            reuseIdentifier: FriendAnnotationView.reuseIdentifier)
+            }
 
-        guard let matchingController = annotationController.first(where: { type(of: annotation) == $0.annotationType }) else {
-            return nil
+            (view as? FriendAnnotationView)?.friend = userAnnotation.friend
+        case .user:
+            if #available(iOS 11.0, *) {
+                if let dequedView = mapView.dequeueReusableAnnotationView(withIdentifier: BikeAnnoationView.reuseIdentifier, for: userAnnotation) as? BikeAnnoationView {
+                    dequedView.annotation = userAnnotation
+                    view = dequedView
+                } else {
+                    view = BikeAnnoationView(annotation: userAnnotation,
+                                             reuseIdentifier: BikeAnnoationView.reuseIdentifier)
+                }
+            } else {
+                view = BikeAnnoationView(annotation: userAnnotation,
+                                         reuseIdentifier: BikeAnnoationView.reuseIdentifier)
+            }
         }
-
-        return mapView.dequeueReusableAnnotationView(ofType: matchingController.annotationViewType, with: annotation)
+        return view
```

Comment:
Maybe we should revert that change and fix the FriendAnnotation in a future PR

Reply:
I'll add a new issue for this 👍

Reply:
https://github.com/criticalmaps/criticalmaps-ios/issues/217